### PR TITLE
Update HISTORY for TileDB-Py 0.30.0 against TileDB 2.24.0

### PR DESCRIPTION
**Pull request by kounelisagis**

### CODE REVIEW

**Comment by jdblischak** [comment]
It appears the Azure jobs all failed to download bzip2. I restarted them

**Comment by jdblischak** [comment]
A restart fixed the libtiledb builds, but now TileDB-Py is failing to import on linux and macOS. And on macOS arm64 it is failing to install the dependencies. They all look related to numpy

```
# linux
    + sh -c 'python -c '"'"'import tiledb'"'"''
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/tmp/tmp.KtRDgSWBXK/venv/lib/python3.8/site-packages/tiledb/__init__.py", line 39, in <module>
    from .array_schema import ArraySchema
  File "/tmp/tmp.KtRDgSWBXK/venv/lib/python3.8/site-packages/tiledb/array_schema.py", line 10, in <module>
    from .attribute import Attr
  File "/tmp/tmp.KtRDgSWBXK/venv/lib/python3.8/site-packages/tiledb/attribute.py", line 10, in <module>
    from .datatypes import DataType
  File "/tmp/tmp.KtRDgSWBXK/venv/lib/python3.8/site-packages/tiledb/datatypes.py", line 197, in <module>
    list(map(DataType.from_numpy, _NUMPY_TO_TILEDB.keys()))
  File "/tmp/tmp.KtRDgSWBXK/venv/lib/python3.8/site-packages/tiledb/datatypes.py", line 21, in from_numpy
    if dtype == "ascii":
TypeError: data type "ascii" not understood

# macOS
+ /bin/sh -c 'python -c '"'"'import tiledb'"'"''
Traceback (most recent call last):
  File "/private/var/folders/v_/hyh3_pjd2rx2qn6110n50l500000gn/T/cibw-run-02zfv0zz/cp38-macosx_x86_64/venv-test-x86_64/lib/python3.8/site-packages/tiledb/__init__.py", line 28, in <module>
    from .libtiledb import Ctx
  File "tiledb/libmetadata.pyx", line 6, in init tiledb.libtiledb
  File "/private/var/folders/v_/hyh3_pjd2rx2qn6110n50l500000gn/T/cibw-run-02zfv0zz/cp38-macosx_x86_64/venv-test-x86_64/lib/python3.8/site-packages/tiledb/datatypes.py", line 197, in <module>
    list(map(DataType.from_numpy, _NUMPY_TO_TILEDB.keys()))
  File "/private/var/folders/v_/hyh3_pjd2rx2qn6110n50l500000gn/T/cibw-run-02zfv0zz/cp38-macosx_x86_64/venv-test-x86_64/lib/python3.8/site-packages/tiledb/datatypes.py", line 21, in from_numpy
    if dtype == "ascii":
TypeError: data type "ascii" not understood

# macOS arm64
  Successfully installed Cython-3.0.10 numpy-1.26.4 pybind11-2.12.0 setuptools-70.0.0 wheel-0.43.0
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Running command Getting requirements to build wheel
  TILEDB_PATH from env: '/Users/runner/work/1/.libtiledb_dist/ff3879bbab2eb2ee8891271dad2d4bcab83dc0f9-macos-arm64'
  TILEDB_PATH in block before: '/Users/runner/work/1/.libtiledb_dist/ff3879bbab2eb2ee8891271dad2d4bcab83dc0f9-macos-arm64'
  TILEDB_PATH in block after: '/Users/runner/work/1/.libtiledb_dist/ff3879bbab2eb2ee8891271dad2d4bcab83dc0f9-macos-arm64'
  libtiledb_exists checking 'library_dirs': ['/Users/runner/work/1/.libtiledb_dist/ff3879bbab2eb2ee8891271dad2d4bcab83dc0f9-macos-arm64/lib']
  libtiledb_exists found: '/Users/runner/work/1/.libtiledb_dist/ff3879bbab2eb2ee8891271dad2d4bcab83dc0f9-macos-arm64/lib/libtiledb.dylib'
  Getting requirements to build wheel: finished with status 'done'
ERROR: Some build dependencies for file:///Users/runner/work/1/s conflict with the backend dependencies: numpy==1.26.4 is incompatible with numpy==1.19.4 ; python_version == "3.9" and platform_machine !='aarch64'.

**Comment by jdblischak** [comment]
Hopefully the numpy problems will be fixed by https://github.com/TileDB-Inc/TileDB-Py/pull/1989

**Comment by kounelisagis** [comment]
> Hopefully the numpy problems will be fixed by #1989

It is: https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=39890&view=results

File: HISTORY.md [hunk 1]
```diff
@@ -1,7 +1,12 @@
-# Release 0.29.1
+# Release 0.30
+
+* TileDB-Py 0.29.0 includes TileDB Embedded [2.24.0](https://github.com/TileDB-Inc/TileDB/releases/tag/2.24.0)
```
Comment (ihnorton) [inline HISTORY.md:3]:
```suggestion
* TileDB-Py 0.30.0 includes TileDB Embedded [2.24.0](https://github.com/TileDB-Inc/TileDB/releases/tag/2.24.0)
```

**Review by ihnorton — APPROVED** [review]

**Pull request merged**